src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt

```diff
@@ -263,8 +265,12 @@ class VisitService(
   fun cancelVisit(reference: String, cancelOutcome: OutcomeDto): VisitDto {
     val visitEntity = visitRepository.findBookedVisit(reference) ?: throw VisitNotFoundException("Visit $reference not found")
 
+    var visitCancellationDateAllowed = LocalDateTime.now()
     // check if the visit being cancelled is in the past
-    validateVisitStartDate(visitEntity, "cancelled")
+    if (visitCancellationDayLimit > 0) {
+      visitCancellationDateAllowed = visitCancellationDateAllowed.minusDays(visitCancellationDayLimit.toLong()).truncatedTo(ChronoUnit.DAYS)
+    }
+    validateVisitStartDate(visitEntity, "cancelled", visitCancellationDateAllowed)
```

Comment:
Would move this into a method.. to encapsulate the complexity and make code easy to read
`    val visitCancellationDateAllowed = getAllowedCancellatonDate()`

Reply:
Agreed, will update.

Reply:
Updated to add getAllowedCancellatonDate method.